sourcegraph/jetbrains
# Cody Ignore: Manual autocomplete trigger notification, sidebar command button sidebar UX and sundry

- Intercepts the manual autocomplete trigger and displays a notification
- Disables the buttons in the sidebar command tab when in an ignored file
- Displays a banner in the sidebar command tab when in an ignored file
- No longer displays banners on individual editors
- Notification text, tooltip text has been updated to match latest UX proposal

https://www.loom.com/share/3a417616ef4f4701ac0e02bcc0c61e64

Part of #1252, #1256

Closes #1253

## Test plan

Manually tested:

- Run plugin with environment `CODY_JETBRAINS_FEATURES=cody.feature.internals-menu=true`
- Use status bar, Internals, Testing: Cody Ignore and change the policy to match your repo, turn the policy override on.
- Check that autocomplete does not happen
- Check that the manual autocomplete trigger displays a notification
- Check that right click menu actions for Explain, Smell and "Test V1" display a notification; same for their keyboard shortcuts
- Check that the buttons in the sidebar command tab are disabled and a banner is displayed
- Check that the status bar Cody icon displays a slash and tooltip when in an ignored file
- Check that all the above and UX affordances respond when the policy changes (override off, different repo name, etc.)
